### DISCUSSION TITLE
CodeQL non-compiled language and --command

### DISCUSSION THREAD

**Discussion by sam-cha-tfs** [github/codeql#13777] [General]
The docs for CodeQL CLI states that using  `--command` with `database create` for a non-compiled language would create an empty database. How would this work if a build script is needed for a non-compiled language?

**Comment by jketema** [comment]
Hi @sam-cha-tfs, thanks for your question. What does the script do? If it generates some of the source files, you would want to run it before creating the database.

**Reply by sam-cha-tfs** [reply]
I'm assuming it's a yes on the empty database. The build script helps setup dependencies.

**Reply by jketema** [reply]
Yes, so if you run the script through `--command` you'll get an empty database. If the script just helps to setup dependencies, just run it _before_ you run `codeql database create`, and run `codeql database create` without the `--command` option. This should create a database that is not empty.

**Comment by sam-cha-tfs** [comment]
For my example I have a python project that takes parameters and throws an error if the parameters are not given. How would I pass these into the project since I can't use the --command?

**Reply by jketema** [reply]
What do you mean by "takes parameters"?

**Reply by sam-cha-tfs** [reply]
arguments are passed into the project example:

`main.py arg1 arg2 .....`

**Reply by jketema** [reply]
My apologies, but I don't understand. Do you need to pass parameters in to the build script, or the program you would eventually run after installing all dependencies?
